contracts/contracts/subnet/SubnetActorManagerFacet.sol

```diff
@@ -296,13 +296,17 @@ contract SubnetActorManagerFacet is SubnetActorModifiers, ReentrancyGuard, Pausa
                 LibSubnetActor.rmAddressFromBalanceKey(msg.sender);
                 s.supplySource.transferFunds(payable(msg.sender), genesisBalance);
             }
+        }
+        // stake might be racing with `bootstrapSubnetIfNeeded` so we need to check again
 
-            // interaction must be performed after checks and changes
+        // interaction must be performed after checks and changes
+        // we do check bootstrapping again, deliberately to avoid a complex exploit scenario
+        if (!s.bootstrapped) {
```

Comment:
Adding this check is to prevent `s.supplySource` call `stake`?

Reply:
I looked into reentrant, but it only protects against a call of the same function. Here two calls interact with the `bootstrapped` flag and hence it's an issue